### ISSUE TITLE
Loop "Is everything correct"

### ISSUE THREAD

**Issue by claudionieder** [vitorgalvao/tiny-scripts#150]
I try to cask-repair intellij-idea-ce and can't get out of the "Is everything correct" question. See output below. Whenever the question appeared I hit the 'y' key and it proceeded the way you see. It continued as soon as I hit the 'y' key and did not wait for the return key to be pressed.

```
$ cask-repair  intellij-idea-ce
Updating taps… Already up-to-date.
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
cask 'intellij-idea-ce' do
  version '2018.2.4,182.4505.22'
  sha256 '22e1f4dffd0576f3c77e47a9de57a56e95e041cb21ce62b28b2454b5eccca2a9'

  url "https://download.jetbrains.com/idea/ideaIC-#{version.before_comma}.dmg"
  appcast 'https://data.services.jetbrains.com/products/releases?code=IIC&latest=true&type=release'
  name 'IntelliJ IDEA Community Edition'
  name 'IntelliJ IDEA CE'
  homepage 'https://www.jetbrains.com/idea/'

  auto_updates true

  app 'IntelliJ IDEA CE.app'

  uninstall_postflight do
    ENV['PATH'].split(File::PATH_SEPARATOR).map { |path| File.join(path, 'idea') }.each { |path| File.delete(path) if File.exist?(path) && File.readlines(path).grep(%r{# see com.intellij.idea.SocketLock for the server side of this interface}).any? }
  end

  zap trash: [
               "~/Library/Application Support/IdeaIC#{version.major_minor}",
               "~/Library/Caches/IdeaIC#{version.major_minor}",
               "~/Library/Logs/IdeaIC#{version.major_minor}",
               "~/Library/Preferences/IdeaIC#{version.major_minor}",
               '~/Library/Preferences/com.jetbrains.intellij.ce.plist',
               '~/Library/Saved Application State/com.jetbrains.intellij.ce.savedState',
             ]
end
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Type the new version (or leave blank to use current one, or use `s` to skip)
> 2018.2.5,182.4892.20
==> Downloading external files for Cask intellij-idea-ce
==> Downloading https://download.jetbrains.com/idea/ideaIC-2018.2.5.dmg
==> Downloading from https://download-cf.jetbrains.com/idea/ideaIC-2018.2.5.dmg
######################################################################## 100.0%
==> No SHA-256 checksum defined for Cask 'intellij-idea-ce', skipping verification.
==> Success! Downloaded to -> /Users/claudioa/Library/Caches/Homebrew/downloads/b56e7998f327b3d3939afb33c47ff3c8b8465b6286b40a809829047ebcfab576--ideaIC-2018.2.5.dmg
diff --git a/Casks/intellij-idea-ce.rb b/Casks/intellij-idea-ce.rb
index 7e1f8ce88c..ba0256840d 100644
--- a/Casks/intellij-idea-ce.rb
+++ b/Casks/intellij-idea-ce.rb
@@ -1,6 +1,6 @@
 cask 'intellij-idea-ce' do
-  version '2018.2.4,182.4505.22'
-  sha256 '22e1f4dffd0576f3c77e47a9de57a56e95e041cb21ce62b28b2454b5eccca2a9'
+  version '2018.2.5,182.4892.20'
+  sha256 '6a04ea48e8d85ac538a119f73e74c6e4292cf9c3dcc7a32dc58c1dea2d40bbed'
 
   url "https://download.jetbrains.com/idea/ideaIC-#{version.before_comma}.dmg"
   appcast 'https://data.services.jetbrains.com/products/releases?code=IIC&latest=true&type=release'
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
• 0 files inspected, no offenses detected
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Is everything correct? ([y]es / [n]o / [e]dit) y
diff --git a/Casks/intellij-idea-ce.rb b/Casks/intellij-idea-ce.rb
index 7e1f8ce88c..ba0256840d 100644
--- a/Casks/intellij-idea-ce.rb
+++ b/Casks/intellij-idea-ce.rb
@@ -1,6 +1,6 @@
 cask 'intellij-idea-ce' do
-  version '2018.2.4,182.4505.22'
-  sha256 '22e1f4dffd0576f3c77e47a9de57a56e95e041cb21ce62b28b2454b5eccca2a9'
+  version '2018.2.5,182.4892.20'
+  sha256 '6a04ea48e8d85ac538a119f73e74c6e4292cf9c3dcc7a32dc58c1dea2d40bbed'
 
   url "https://download.jetbrains.com/idea/ideaIC-#{version.before_comma}.dmg"
   appcast 'https://data.services.jetbrains.com/products/releases?code=IIC&latest=true&type=release'
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Is everything correct? ([y]es / [n]o / [e]dit) y
diff --git a/Casks/intellij-idea-ce.rb b/Casks/intellij-idea-ce.rb
index 7e1f8ce88c..ba0256840d 100644
--- a/Casks/intellij-idea-ce.rb
+++ b/Casks/intellij-idea-ce.rb
@@ -1,6 +1,6 @@
 cask 'intellij-idea-ce' do
-  version '2018.2.4,182.4505.22'
-  sha256 '22e1f4dffd0576f3c77e47a9de57a56e95e041cb21ce62b28b2454b5eccca2a9'
+  version '2018.2.5,182.4892.20'
+  sha256 '6a04ea48e8d85ac538a119f73e74c6e4292cf9c3dcc7a32dc58c1dea2d40bbed'
 
   url "https://download.jetbrains.com/idea/ideaIC-#{version.before_comma}.dmg"
   appcast 'https://data.services.jetbrains.com/products/releases?code=IIC&latest=true&type=release'
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Is everything correct? ([y]es / [n]o / [e]dit) y
diff --git a/Casks/intellij-idea-ce.rb b/Casks/intellij-idea-ce.rb
index 7e1f8ce88c..ba0256840d 100644
--- a/Casks/intellij-idea-ce.rb
+++ b/Casks/intellij-idea-ce.rb
@@ -1,6 +1,6 @@
 cask 'intellij-idea-ce' do
-  version '2018.2.4,182.4505.22'
-  sha256 '22e1f4dffd0576f3c77e47a9de57a56e95e041cb21ce62b28b2454b5eccca2a9'
+  version '2018.2.5,182.4892.20'
+  sha256 '6a04ea48e8d85ac538a119f73e74c6e4292cf9c3dcc7a32dc58c1dea2d40bbed'
 
   url "https://download.jetbrains.com/idea/ideaIC-#{version.before_comma}.dmg"
   appcast 'https://data.services.jetbrains.com/products/releases?code=IIC&latest=true&type=release'
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Is everything correct? ([y]es / [n]o / [e]dit) y
diff --git a/Casks/intellij-idea-ce.rb b/Casks/intellij-idea-ce.rb
index 7e1f8ce88c..ba0256840d 100644
--- a/Casks/intellij-idea-ce.rb
+++ b/Casks/intellij-idea-ce.rb
@@ -1,6 +1,6 @@
 cask 'intellij-idea-ce' do
-  version '2018.2.4,182.4505.22'
-  sha256 '22e1f4dffd0576f3c77e47a9de57a56e95e041cb21ce62b28b2454b5eccca2a9'
+  version '2018.2.5,182.4892.20'
+  sha256 '6a04ea48e8d85ac538a119f73e74c6e4292cf9c3dcc7a32dc58c1dea2d40bbed'
 
   url "https://download.jetbrains.com/idea/ideaIC-#{version.before_comma}.dmg"
   appcast 'https://data.services.jetbrains.com/products/releases?code=IIC&latest=true&type=release'
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Is everything correct? ([y]es / [n]o / [e]dit) y
diff --git a/Casks/intellij-idea-ce.rb b/Casks/intellij-idea-ce.rb
index 7e1f8ce88c..ba0256840d 100644
--- a/Casks/intellij-idea-ce.rb
+++ b/Casks/intellij-idea-ce.rb
@@ -1,6 +1,6 @@
 cask 'intellij-idea-ce' do
-  version '2018.2.4,182.4505.22'
-  sha256 '22e1f4dffd0576f3c77e47a9de57a56e95e041cb21ce62b28b2454b5eccca2a9'
+  version '2018.2.5,182.4892.20'
+  sha256 '6a04ea48e8d85ac538a119f73e74c6e4292cf9c3dcc7a32dc58c1dea2d40bbed'
 
   url "https://download.jetbrains.com/idea/ideaIC-#{version.before_comma}.dmg"
   appcast 'https://data.services.jetbrains.com/products/releases?code=IIC&latest=true&type=release'
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Is everything correct? ([y]es / [n]o / [e]dit) y
diff --git a/Casks/intellij-idea-ce.rb b/Casks/intellij-idea-ce.rb
index 7e1f8ce88c..ba0256840d 100644
--- a/Casks/intellij-idea-ce.rb
+++ b/Casks/intellij-idea-ce.rb
@@ -1,6 +1,6 @@
 cask 'intellij-idea-ce' do
-  version '2018.2.4,182.4505.22'
-  sha256 '22e1f4dffd0576f3c77e47a9de57a56e95e041cb21ce62b28b2454b5eccca2a9'
+  version '2018.2.5,182.4892.20'
+  sha256 '6a04ea48e8d85ac538a119f73e74c6e4292cf9c3dcc7a32dc58c1dea2d40bbed'
 
   url "https://download.jetbrains.com/idea/ideaIC-#{version.before_comma}.dmg"
   appcast 'https://data.services.jetbrains.com/products/releases?code=IIC&latest=true&type=release'
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Is everything correct? ([y]es / [n]o / [e]dit) y
diff --git a/Casks/intellij-idea-ce.rb b/Casks/intellij-idea-ce.rb
index 7e1f8ce88c..ba0256840d 100644
--- a/Casks/intellij-idea-ce.rb
+++ b/Casks/intellij-idea-ce.rb
@@ -1,6 +1,6 @@
 cask 'intellij-idea-ce' do
-  version '2018.2.4,182.4505.22'
-  sha256 '22e1f4dffd0576f3c77e47a9de57a56e95e041cb21ce62b28b2454b5eccca2a9'
+  version '2018.2.5,182.4892.20'
+  sha256 '6a04ea48e8d85ac538a119f73e74c6e4292cf9c3dcc7a32dc58c1dea2d40bbed'
 
   url "https://download.jetbrains.com/idea/ideaIC-#{version.before_comma}.dmg"
   appcast 'https://data.services.jetbrains.com/products/releases?code=IIC&latest=true&type=release'
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
• 0 files inspected, no offenses detected
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Is everything correct? ([y]es / [n]o / [e]dit) 

```

**Comment by MindTooth** [comment]
Hi,

Care to share what you're trying to achive?  The new version is already submitted [here](https://github.com/Homebrew/homebrew-cask/commit/39790d1cd7c5760fe8a3a942d177ca246562b5c6).

**Comment by vitorgalvao** [comment]
Can you `brew cask style` correctly?

**Comment by claudionieder** [comment]
@vitorgalvao:

```
$ brew cask style

5589 files inspected, no offenses detected
```

@MindTooth:

At the time I executed the command I did not know yet that the version 2018.2.5 was already available. Later on the same day I could update it. Anyway as this looping behaviour is strange I thought better report it.

**Comment by vitorgalvao** [comment]
@claudionieder Can you reproduce this with another cask?

**Comment by josser** [comment]
I can reproduce it with decrediton
However, my cask style is not good: 

```
$ brew cask style
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- jaro_winkler (LoadError)
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.3.0/gems/rubocop-0.59.2/lib/rubocop/string_util.rb:3:in `<top (required)>'
	from /usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.3.0/gems/rubocop-0.59.2/lib/rubocop.rb:21:in `require_relative'
	from /usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.3.0/gems/rubocop-0.59.2/lib/rubocop.rb:21:in `<top (required)>'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.3.0/gems/rubocop-0.59.2/exe/rubocop:6:in `<top (required)>'
	from /usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.3.0/bin/rubocop:22:in `load'
	from /usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.3.0/bin/rubocop:22:in `<main>'
Error: style check failed

```

**Comment by josser** [comment]
@MindTooth Hm .. 
```
$ /usr/bin/gem --list
ERROR:  Invalid option: --list.  See 'gem --help'.
```

**Comment by josser** [comment]
```
$ /usr/bin/gem list  

*** LOCAL GEMS ***

bigdecimal (1.2.8)
CFPropertyList (2.2.8)
did_you_mean (1.0.0)
io-console (0.4.5)
json (1.8.3.1)
libxml-ruby (2.9.0)
minitest (5.8.5)
net-telnet (0.1.1)
nokogiri (1.5.6)
power_assert (0.2.6)
psych (2.1.0.1)
rake (10.4.2)
rdoc (4.2.1)
sqlite3 (1.3.11)
test-unit (3.1.5)
```

**Comment by vitorgalvao** [comment]
Going to close because `style`-related problems are not the domain of `cask-repair`. But feel free to continue the debugging in this thread.

**Comment by josser** [comment]
It doesn't look like issue with style is related to infinite loop. 
@vitorgalvao as I understand you can't reproduce issue? 
What is you env? Can't it be related to new os x mojave? I'm also on zsh.

**Comment by vitorgalvao** [comment]
> Can't it be related to new os x mojave?

No. Tons of people are on it and don’t have the issue.

> AS I understand, when cask-repair asking for 'is everything correct' it call cask style after.

No, it calls it before. But it won’t continue if there are certain types of errors.

**Comment by claudionieder** [comment]
Sorry for not answering earlier, I was busy :-(

Just in case it can help here my output of /usr/bin/gem list
```
*** LOCAL GEMS ***

ast (2.3.0)
bigdecimal (1.2.8)
CFPropertyList (2.2.8)
did_you_mean (1.0.0)
io-console (0.4.5)
json (1.8.3.1)
libxml-ruby (2.9.0)
minitest (5.8.5)
net-telnet (0.1.1)
nokogiri (1.5.6)
parallel (1.12.1)
parser (2.4.0.2)
power_assert (0.2.6)
powerpack (0.1.1)
psych (2.1.0.1)
public_suffix (3.0.1)
rainbow (2.2.2)
rake (10.4.2)
rdoc (4.2.1)
rubocop (0.51.0)
rubocop-cask (0.15.1)
ruby-progressbar (1.9.0)
sqlite3 (1.3.11)
test-unit (3.1.5)
unicode-display_width (1.3.0)
```
At the moment I cannot reproduce the behaviour, because it now says:

No changes made to the cask. Skipping…

As @MindTooth mentioned, at the time I tried, IntelliJ 2018.2.5 was already available, I just didn't know, so could that be the reason for the strange behaviour, i.e. that I was not up to date with what was in the homwbrew-cask repository?

**Comment by neverfox** [comment]
I'm having this issue as well (which just so happened to seem to occur around the time I updated to Mojave and when an update to cask-repair landed, though I acknowledge that those aren't suspected as reasons).

```
❯ /usr/bin/gem list

*** LOCAL GEMS ***

bigdecimal (1.2.8)
CFPropertyList (2.2.8)
did_you_mean (1.0.0)
io-console (0.4.5)
json (1.8.3.1)
libxml-ruby (2.9.0)
minitest (5.8.5)
net-telnet (0.1.1)
nokogiri (1.5.6)
power_assert (0.2.6)
psych (2.1.0.1)
rake (10.4.2)
rdoc (4.2.1)
sqlite3 (1.3.11)
test-unit (3.1.5)
```

@josser What was your solution exactly?

**Comment by josser** [comment]
@neverfox 
Here is command:
```
brew vendor-install ruby
```

**Comment by MindTooth** [comment]
I don't think `did_you_mean` is part of the base system.  And if I recall; I did run a `sudo /usr/bin/gem uninstall did_you_mean` at one time.  And it worked somehow afterwards.

The reason I asked for the gem list; is that I know for sure that one gem were installed as root, and it interfered with `brew (cask) style`.  After I removed one(more) system gem(s), I were able to fix my issue.

Regarding installing the vendor Ruby; that broke each time Homebrew had an update — resulting in a broken system once more.

**Comment by josser** [comment]
@MindTooth but did_you_mean is in local list, isn't it? 
I've never used sudo to install something on my system.

**Comment by MindTooth** [comment]
> @MindTooth but did_you_mean is in local list, isn't it?
> I've never used sudo to install something on my system.
I did plan on bringing a clean Mojave up in VirtualBox to see which gems is installed.

Thought I would only share my solution.  Also I did never install gems explicitly via `sudo` that I know of.  Don't really understand why removing the ones `brew` complained about solved it.
I do suspect that it might be because the gem is installed as a system gem, and also it don't meet the requirement of `brew`; therefore it's unable to upgrade said gem, and crashes.  Just my two cents..
